src/clients/TorrentClient.ts

```diff
@@ -1,25 +1,51 @@
 import { Metafile } from "../parseTorrent.js";
-import { DecisionAnyMatch, InjectionResult } from "../constants.js";
+import {
+	Decision,
+	DecisionAnyMatch,
+	InjectionResult,
+	VIDEO_DISC_EXTENSIONS,
+} from "../constants.js";
 import { getRuntimeConfig } from "../runtimeConfig.js";
 import { Searchee, SearcheeWithInfoHash } from "../searchee.js";
 import QBittorrent from "./QBittorrent.js";
 import RTorrent from "./RTorrent.js";
 import Transmission from "./Transmission.js";
 import Deluge from "./Deluge.js";
 import { Result } from "../Result.js";
+import { Label } from "../logger.js";
+import { hasExt } from "../utils.js";
 
-let activeClient: TorrentClient;
+let activeClient: TorrentClient | null = null;
+
+export type TorrentClientType =
+	| Label.QBITTORRENT
+	| Label.RTORRENT
+	| Label.TRANSMISSION
+	| Label.DELUGE;
+
+export interface GenericTorrentInfo {
```

Comment:
I'm trying to think of something slightly more descriptive but nothing is great…`TorrentMetadata`? `TorrentMetadataInClient` `ClientTorrentMetadata` `TorrentClientMetadata` `TorrentAttributes` `TorrentParameters` `ClientTorrentParameters` etc. 

idk 🤷